### PR TITLE
James 3308 - add test for DistributedTaskManagerTest

### PR DESCRIPTION
**Pull request by nvduc91**
Add test for DistributedTaskManagerTest: make sure taskManager working fine with error when deserialization

### CODE REVIEW

File: server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java [hunk 1]
```diff
@@ -196,6 +202,86 @@ EventSourcingTaskManager taskManager(Hostname hostname) {
         return new EventSourcingTaskManager(workQueueSupplier, eventStore, executionDetailsProjection, hostname, terminationSubscriber);
     }
 
+    @Test
+    void badPayLoadShouldNotAffectToTaskManagerOnCancelTask() throws TaskManager.ReachedTimeoutException {
```
Comment (Arsnael) [inline server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java:206]:
```suggestion
    void badPayloadShouldNotAffectTaskManagerOnCancelTask() throws TaskManager.ReachedTimeoutException {
```

File: server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java [hunk 2]
```diff
@@ -196,6 +202,86 @@ EventSourcingTaskManager taskManager(Hostname hostname) {
...
+            .block();
+
+        taskManager.cancel(id);
+        taskManager.await(id, TIMEOUT);
+        assertThat(taskManager.getExecutionDetails(id).getStatus())
+            .isEqualTo(TaskManager.Status.CANCELLED);
+    }
+
+    @Test
+    void badPayLoadsShouldNotAffectToTaskManagerOnCancelTask() throws TaskManager.ReachedTimeoutException {
```
Comment (Arsnael) [inline server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java:226]:
```suggestion
    void badPayloadsShouldNotAffectTaskManagerOnCancelTask() throws TaskManager.ReachedTimeoutException {
```

File: server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java [hunk 3]
```diff
@@ -196,6 +202,86 @@ EventSourcingTaskManager taskManager(Hostname hostname) {
...
+            .block());
+
+        taskManager.cancel(id);
+        taskManager.await(id, TIMEOUT);
+        assertThat(taskManager.getExecutionDetails(id).getStatus())
+            .isEqualTo(TaskManager.Status.CANCELLED);
+    }
+
+    @Test
+    void badPayLoadShouldNotAffectToTaskManagerOnCompleteTask() throws TaskManager.ReachedTimeoutException {
```
Comment (Arsnael) [inline server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java:247]:
```suggestion
    void badPayloadShouldNotAffectTaskManagerOnCompleteTask() throws TaskManager.ReachedTimeoutException {
```

File: server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java [hunk 4]
```diff
@@ -196,6 +202,86 @@ EventSourcingTaskManager taskManager(Hostname hostname) {
...
+                "BAD_PAYLOAD!".getBytes(UTF_8))))
+            .block();
+
+        taskManager.await(id, TIMEOUT);
+        assertThat(taskManager.getExecutionDetails(id).getStatus())
+            .isEqualTo(TaskManager.Status.COMPLETED);
+    }
+
+    @Test
+    void badPayLoadsShouldNotAffectToTaskManagerOnCompleteTask() throws TaskManager.ReachedTimeoutException {
```
Comment (Arsnael) [inline server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java:266]:
```suggestion
    void badPayloadsShouldNotAffectTaskManagerOnCompleteTask() throws TaskManager.ReachedTimeoutException {
```

File: server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java (hunk 1 above)
Comment (chibenwa) [inline server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java:206]: s/PayLoad/Payload

File: server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java (hunk 2 above)
Comment (chibenwa) [inline server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java:226]: s/PayLoad/Payload

File: server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java (hunk 3 above)
Comment (chibenwa) [inline server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java:247]: idem

File: server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java (hunk 4 above)
Comment (chibenwa) [inline server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java:266]: idem

**Comment by chibenwa** [comment]
```
[014fd887d543d9b8a5ca8a6dff50ec812b0bd2bb] [ERROR]   MemorySpamAssassinContractTest.deletingSpamMessageShouldNotImpactSpamAssassinState » ConditionTimeout
[014fd887d543d9b8a5ca8a6dff50ec812b0bd2bb] [ERROR]   MemorySpamAssassinContractTest.spamShouldBeDeliveredInSpamMailboxWhenSameMessageHasAlreadyBeenMovedToSpam » ConditionTimeout
```

test this please

**Comment by chibenwa** [comment]
(oups, a rebase is needed...)

**Comment by nvduc91** [comment]
rebase master

**Comment by nvduc91** [comment]
test this please

**Comment by nvduc91** [comment]
test this please

```
Failed to execute goal on project james-server-util: Could not resolve dependencies for project org.apache.james:james-server-util:jar:3.6.0-SNAPSHOT: Failed to collect dependencies at com.github.steveash.guavate:guavate:jar:1.0.0 -> com.google.guava:guava:jar:25.1-jre -> org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Failed to read artifact descriptor for org.codehaus.mojo:animal-sniffer-annotations:jar:1.14: Could not transfer artifact org.codehaus.mojo:animal-sniffer-annotations:pom:1.14 from/to central (https://repo.maven.apache.org/maven2): /root/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.pom.part (No such file or directory) -> [Help 1]
```

**Comment by chibenwa** [comment]
`Caused by: java.io.FileNotFoundException: dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/blob.properties (No such file or directory)`

Idem. Is this fixed?

test this please

**Comment by chibenwa** [comment]
`Caused by: java.io.FileNotFoundException: dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/blob.properties (No such file or directory)`

Should be fixed now

test this please

**Comment by chibenwa** [comment]
`Caused by: java.io.FileNotFoundException: null/dockerfiles/run/guice/cassandra-rabbitmq-ldap/destination/conf/blob.properties (No such file or directory)`

test this please

Can be considered :green_apple:

**Comment by chibenwa** [comment]
test this please

**Comment by blackheaven** [comment]
test this please

**Comment by chibenwa** [comment]
```
RabbitMQDeletedMessageVaultIntegrationTest>DeletedMessageVaultIntegrationTest.vaultExportShouldExportZipContainsVaultMessagesToShareeWhenImapDeletedMailbox:543 
[23d96a632892cf50d0f62464211acbf866c4bdc6] Expecting zipFile to contains 1 entries but actually contains (0) entries
```

test this please

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**